File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/Model/IDONotificationStatusParamModel.swift
//
//  IDONotificationStatusParamModel.swift
//  protocol_channel
//
//  Created by hc on 2023/10/12.
//

import Foundation
/// Notification app status setting event
@objcMembers
public class IDONotificationStatusParamModel: NSObject, IDOBaseModel {
    /// Notification type:
    /// 1: Allow notification
    /// 2: Silent notification
    /// 3: Disable notification
    public var notifyFlag: Int

    enum CodingKeys: String, CodingKey {
        case notifyFlag = "notify_flag"
    }

    public init(notifyFlag: Int) {
        self.notifyFlag = notifyFlag
    }

    public func toJsonString() -> String? {
        if let jsonData = try? JSONEncoder().encode(self) {
            let jsonString = String(data: jsonData, encoding: .utf8)
            return jsonString
        }
        return nil
    }
}

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/model/IDONotificationStatusParamModel.kt
//
//  IDONotificationStatusParamModel.kt
//  protocol_channel
//
//  Created by hedongyang on 2023/10/30.
//

package com.idosmart.model

import com.google.gson.GsonBuilder
import com.google.gson.annotations.SerializedName

///
/// Notification app status setting event

open class IDONotificationStatusParamModel(notifyFlag: Int) : IDOBaseModel {

    /// Notification type:
    /// 1: Allow notification
    /// 2: Silent notification
    /// 3: Disable notification
    @SerializedName("notify_flag")
    var notifyFlag: Int = notifyFlag

    

    override fun toJsonString(): String {
        return GsonBuilder().create().toJson(this).toString()
    }
}
    